File: Warehousing.Data/Entities/RolePermission.cs
namespace Warehousing.Data.Entities
{
    public class RolePermission : BaseClass
    {
        public int Id { get; set; }
        public Role? Role { get; set; }
        public int RoleId { get; set; }

        public Permission? Permission { get; set; }
        public int PermissionId { get; set; }
    }
}

File: Warehousing.UI/src/app/admin/models/rolePermission.ts
import { Permission } from "./permission";
import { Role } from "./role";

export interface RolePermission {
    id: number;
    role: Role;
    roleId: number;
    permission: Permission;
    permissionId: number;
}

export interface RolePermissionDto {
    id: number;
    roleId: number;
    permissionId: number;
    nameEn: string;
    nameAr: string;
    code: string;
}